🐛 Pre-Release version does not work with Biome v2 or Biome v1

### VS Code version

1.98.2

### Extension version

2025.3.151143

### Biome version

2.0.0-beta.1 (none displayed in the statusbar)

### Operating system

- [x] Windows
- [ ] macOS
- [ ] Linux

### Description

When installing Biome 2.0.0-beta.1 (or Biome 1.9.4) and the pre-release version of the extension as recommended by the Biome 2 beta blog post, the extension does not work correctly. 
- It can't be registered as a formatter and does not show any linter errors. 
- When trying to use the "Biome: Download CLI" option it does not show 2.0.0-beta.1 as an option. Trying to select any other version throws the error "Cannot find module './index-CJAZtKJE.js'. 
- The icon in the bottom right shows a check-mark but does not show a version and there is no popup asking to install a biome version either.
These issues also persisted when switching back to Biome 1.9.4, or doing a reinstall with 1.9.4 and using the pre-release extension. 
The non pre-release extension seems to work fine with Biome 1.9.4

### Steps to reproduce

1. Add Biome 2.0.0-beta.1 to project (Or Biome 1.9.4)
2. Install Biome extension in pre-release mode
3. Try using the extension

### Expected behavior

The extension should be able to be registered as a formatter and should show linter error. 
The Biome: Download CLI option should not throw an error when used.
Indicator on bottom right should show a biome version

### Does this issue occur when using the CLI directly?

No

### Link to a minimal reproduction
https://github.com/sinomsinom/biome-repro-1743157065790
This issue is in any repo I tried (including an empty repo created with the npm create @biomejs/biome-reproduction command) so a repro repo doesn't seem useful here

### Logs

```plain text
Output:
2025-03-28 10:35:26.190 [info] 





2025-03-28 10:35:26.190 [info] Biome extension 2025.3.151143 activated
2025-03-28 10:35:26.190 [info] Biome extension started
2025-03-28 10:35:26.190 [info] User-facing commands registered
2025-03-28 10:35:26.190 [info] Started listening for lockfile changes
2025-03-28 10:35:26.190 [info] Started listening for configuration changes
2025-03-28 10:35:26.190 [info] Started listening for active text editor changes



biome rage --daemon-logs:
CLI:
  Version:                      2.0.0-beta.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         C:\Users\<path-to-project>\biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0

Discovering running Biome servers...
```

This config issue probably has nothing to do with this issue. 

I don't get any error whatsoever and I made sure the config would be correct. Using the command line interface I can use biome 1 and 2 no problem with the corresponding configs. It only happens if I try to use the plugin with a version newer than https://github.com/biomejs/biome-vscode/commit/602c510201c59d072315769728050c94152889c3 which makes me think it might be an issue where the extension just can't find the workspace or any files in it at all for some reason? But I also can't find any diagnostic messages, errors, or anything that would say as much.

With any commit before that specific one the biome icon in the ribbon at the bottom also correctly shows the used biome version, however with that commit and newer ones it does not while still showing the "up and running" on hover. This makes me think it might also not be able to find the biome LSP (or potentially just unable to find biome in general), but again there is no error message to indicate this. The extension also doesn't even report parsing errors. It basically just doesn't do anything while in that state.

The only error message I get is when trying to download a biome version through the "Biome: Download CLI" option which throws a "Cannot find module './index-CJAZtKJE.js'" error. This is also not an issue with versions older than https://github.com/biomejs/biome-vscode/commit/602c510201c59d072315769728050c94152889c3

Here's an image of that error message:
![Image](https://github.com/user-attachments/assets/7b434dcc-fb61-4e2c-adda-a8549c7b8bb1)

--- a/src/binary-finder.ts
+++ b/src/binary-finder.ts
@@ -67,11 +67,7 @@ const vsCodeSettingsStrategy = {
 				return;
 			}
 
-			const resolvedBinPath = path
-				? Utils.resolvePath(path, bin).toString()
-				: bin;
-
-			const biome = Uri.parse(resolvedBinPath);
+			const biome = path ? Utils.resolvePath(path, bin) : Uri.file(bin);
 
 			if (await fileExists(biome)) {
 				return biome;

--- a/src/binary-finder.ts
+++ b/src/binary-finder.ts
@@ -2,7 +2,6 @@ import { createRequire } from "node:module";
 import { dirname, join } from "node:path";
 import { delimiter } from "node:path";
 import { Uri, window } from "vscode";
-import { Utils } from "vscode-uri";
 import { config, getLspBin } from "./config";
 import {
 	platformIdentifier,
@@ -67,7 +66,7 @@ const vsCodeSettingsStrategy = {
 				return;
 			}
 
-			const biome = path ? Utils.resolvePath(path, bin) : Uri.file(bin);
+			const biome = Uri.file(bin);
 
 			if (await fileExists(biome)) {
 				return biome;

--- a/src/project.ts
+++ b/src/project.ts
@@ -103,7 +103,7 @@ const getProjectDefinitionForSingleFile = (): ProjectDefinition | undefined => {
 		return;
 	}
 
-	const parentFolderURI = Uri.parse(
+	const parentFolderURI = Uri.file(
 		Utils.resolvePath(singleFileURI, "..").fsPath,
 	);
 

--- a/src/utils.ts
+++ b/src/utils.ts
@@ -94,7 +94,7 @@ export const subtractURI = (original: Uri, subtract: Uri): Uri | undefined => {
 
 	result = result === "" ? "/" : result;
 
-	return Uri.parse(result);
+	return Uri.file(result);
 };
 
 /**
@@ -242,7 +242,7 @@ export const getPathRelativeToWorkspaceFolder = (
 	folder: WorkspaceFolder,
 	path?: string,
 ): Uri => {
-	return Uri.parse(Utils.joinPath(folder.uri, path ?? "").fsPath);
+	return Uri.file(Utils.joinPath(folder.uri, path ?? "").fsPath);
 };
 
 /**

